src/theory/strings/core_solver.h

```diff
@@ -219,6 +217,45 @@ class CoreSolver
    */
   Node getNormalString(Node x, std::vector<Node>& nf_exp);
   //-------------------------- end query functions
+
+  /**
+   * This returns the conclusion of the proof rule corresponding to splitting
+   * on the arrangement of terms t and s appearing in an equation of the form
+   *   t ++ t' = s ++ s' or t' ++ t = s' ++ s
+   * where we are in the second case if isRev is true. This method is called
+   * both by the core solver and by the strings proof checker.
+   *
+   * @param t The first term
+   * @param s The second term
```

Comment:
This is not really true for the `STRING_DECOMPOSE` rule. Maybe we should use different symbols in the equations above and make it clear that `t` and `s` are the characteristic terms for that equation + rule and not necessarily the terms in the equation themselves?

Reply:
I'll just split to a new utility method, since indeed STRING_DECOMPOSE doesn't fit this pattern very well.